packages/core/src/components/search/Filter/FilterDesktop.tsx

```diff
@@ -46,65 +50,110 @@ function FilterDesktop({
         dispatch({ type: 'toggleExpanded', payload: idx })
       }
     >
-      {facets.map((facet, index) => {
+      {filteredFacets.map((facet, index) => {
         const { __typename: type, label } = facet
         const isExpanded = expanded.has(index)
         return (
-          <UIFilterFacets
-            key={`${testId}-${label}-${index}`}
-            testId={testId}
-            index={index}
-            type={type}
-            label={label}
-          >
-            {type === 'StoreFacetBoolean' && isExpanded && (
-              <UIFilterFacetBoolean>
-                {facet.values.map((item) => (
-                  <UIFilterFacetBooleanItem
-                    key={`${testId}-${facet.label}-${item.label}`}
-                    id={`${testId}-${facet.label}-${item.label}`}
-                    testId={testId}
-                    onFacetChange={(facet) => {
-                      setState({
-                        ...state,
-                        selectedFacets: toggleFacet(
-                          state.selectedFacets,
-                          facet
-                        ),
-                        page: 0,
-                      })
-                      resetInfiniteScroll(0)
-                    }}
-                    selected={item.selected}
-                    value={item.value}
-                    quantity={item.quantity}
-                    facetKey={facet.key}
-                    label={item.label}
-                  />
-                ))}
-              </UIFilterFacetBoolean>
-            )}
-            {type === 'StoreFacetRange' && isExpanded && (
-              <UIFilterFacetRange
-                facetKey={facet.key}
-                min={facet.min}
-                max={facet.max}
-                formatter={
-                  facet.key.toLowerCase() === 'price'
-                    ? useFormattedPrice
-                    : undefined
-                }
-                onFacetChange={(facet) => {
-                  setState({
-                    ...state,
-                    selectedFacets: setFacet(state.selectedFacets, facet, true),
-                    page: 0,
-                  })
-                  resetInfiniteScroll(0)
-                }}
-              />
+          <>
+            {deliveryPromise.enabled && !postalCode && (
+              <UIFilterFacets
+                key={`${testId}-delivery-unset`}
+                testId={testId}
+                index={index - 1}
```

Comment:
maybe not the best approach, but I think it works. Adding an index-1 just to "ignore" this facet. No zipCode = no shipping facet.